### DISCUSSION TITLE
Best practice analyzer fails to apply fixes when connected to a Power BI model

### DISCUSSION THREAD

**Discussion by samusaran** [TabularEditor/TabularEditor3#1056] [Ideas]
As title, if I use the best practice analyzer, no "Apply fixes" really works, but it forces me to apply the fix manually.

It do work when working on a model.bim, both online and offline.

**Comment by relax528** [comment]
Same when connected to Power BI model.

**Comment by otykier** [comment]
What kind of fixes are we talking about?

Keep in mind that Tabular Editor by default blocks all [modeling operations that are not supported by Power BI Desktop](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#supported-write-operations). So if a best practice rule tries to change a column property, for example, then it will not work, unless you explicitly allow unsupported Power BI modeling in Tabular Editor 3. To do this go to Tools > Preferences > Power BI > (Check) Allow unsupported modeling operations. Keep in mind this may break your .pbix file, so keep a backup just in case!

**Comment by samusaran** [comment]
That's nice. The issue then is that Tabular Editor does not make it clear that it's blocking the operation. The button "apply fix" is active, but it just does not do anything when clicked.
________________________________
Da: Daniel Otykier ***@***.***>
Inviato: mercoledì 4 agosto 2021 15:14
A: TabularEditor/TabularEditor3
Cc: Alessandro Losi; Author
Oggetto: Re: [TabularEditor/TabularEditor3] Best practice analyzer fails to apply fixes when connected to a Power BI model (#151)


What kind of fixes are we talking about?

Keep in mind that Tabular Editor by default blocks all modeling operations that are not supported by Power BI Desktop<https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#supported-write-operations>. So if a best practice rule tries to change a column property, for example, then it will not work, unless you explicitly allow unsupported Power BI modeling in Tabular Editor 3. To do this go to Tools > Preferences > Power BI > (Check) Allow unsupported modeling operations. Keep in mind this may break your .pbix file, so keep a backup just in case!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/TabularEditor/TabularEditor3/issues/151#issuecomment-892648234>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AACG7NZ6LLRHIX5LYORYNGLT3E4MDANCNFSM5BDZXPTQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.
